src/main/java/com/google/codeu/servlets/ImageHandlerServlet.java

```diff
@@ -0,0 +1,134 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.util.List;
+
+import java.util.Map;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.User;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload
+ * and then forwards the request to this servlet. This servlet can then
+ * process the request using the file URL we get from Blobstore.
+ */
+@WebServlet("/image-form-handler")
+public class ImageHandlerServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+
+    // Output some HTML that shows the data the user entered.
+    // A real codebase would probably store these in Datastore.
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    User userData = datastore.getUser(userEmail);
+    String aboutMe;
+
+    if (userData == null || userData.getAboutMe() == null) {
+      aboutMe = "";
+    } else{
+      aboutMe = userData.getAboutMe();
+    }
+
+    String profilePicUrl = imageUrl;
+
+    User user = new User(userEmail, aboutMe, profilePicUrl);
+    datastore.storeUser(user);
```

Comment:
I am not sure how datastore handles this, does it create a new user or update the old user?

Reply:
It will update the old user! From what I understand, it uses the user email as a sort of key which will update it if the about me information or profile pic url changes. I have tried it with our deployed website and monitored the cloud console so should be about right